spring-jdbc/src/main/java/org/springframework/jdbc/datasource/init/ResourceDatabasePopulator.java

```diff
@@ -171,9 +171,22 @@ public void setSeparator(String separator) {
 	 * Set the prefix that identifies single-line comments within the SQL scripts.
 	 * <p>Defaults to {@code "--"}.
 	 * @param commentPrefix the prefix for single-line comments
+	 * @see #setCommentPrefixes(String...)
 	 */
 	public void setCommentPrefix(String commentPrefix) {
-		this.commentPrefix = commentPrefix;
+		Assert.hasText(commentPrefix, "CommentPrefix must not be null or empty");
+		this.commentPrefixes = new String[] { commentPrefix };
+	}
+
+	/**
+	 * Set the prefixes that identify single-line comments within the SQL scripts.
+	 * <p>Defaults to {@code "--"}.
+	 * @param commentPrefixes the prefixes for single-line comments
+	 * @since 5.2
+	 */
+	public void setCommentPrefixes(String... commentPrefixes) {
+		Assert.notNull(commentPrefixes, "CommentPrefixes must not be null");
```

Comment:
```suggestion
		Assert.notEmpty(commentPrefixes, "'commentPrefixes' must not be null or empty");
		Assert.noNullElements(commentPrefixes, "'commentPrefixes' must not contain null elements");
```

Reply:
Perhaps it's better to re-use/re-implement the "no null or empty elements" check you added in `splitSqlScript`.

Reply:
That will sort of happen indirectly I think when the script runs. Won't hurt to also do it early here.